### PR TITLE
Live delta-to-best during the lap

### PR DESCRIPTION
**Pull request by jbhoorasingh**
## What

The delta widget previously only updated when a lap completed (`last_lap_ms - prev_best_ms`). It now ticks **live during the lap**: the gap between your current track position and where the session-best lap was at the same point.

## How

- `TelemetryService` retains the session-best lap's `(dist, t)` trace when a `CompletedLap` becomes the best (safe by reference — the processor re-allocates its sample store at every lap boundary). Cleared on session change.
- Every live frame computes `delta_ms = t_now - t_ref(dist_now)` using a new `analysis.time_delta_at()` — the single-point form of the interpolation the Analysis view's delta chart already uses (positive = slower). Frames also gain `lap_elapsed_ms`.
- `delta_ms` is `null` (widget shows the fallback) when: no reference lap yet, paused/off-track, or the current lap has run past the reference's final sample (edge clamping would otherwise inflate the gap 1 ms/ms).
- Frontend: `liveDelta()` helper in `widgets/shared.tsx` — live gap when available, else the old end-of-lap comparison labeled *Δ best (last lap)*. Used by DeltaWidget (both variants) and the Live view card. Demo mode fakes a swinging delta so both colors show.

## Testing

- 6 new backend tests (`test_live_delta.py`): pure-helper sign/interpolation/edge cases, plus service-level tests driving laps at different speeds — faster lap → negative delta, slower → positive, null before a reference / past its end / after a session reset. Full suite: 74 passed; ruff + strict mypy clean.
- `npm run lint && npm run build` clean.
- Verified against the simulator: lap 1 shows "–", from lap 2 the delta ticks continuously (observed +0.377 → +0.452 → +0.502 over 10 s on a slower lap, matching the sim's per-lap pace jitter).

## Docs

Overlay widget table, dash guide (live-delta explanation), and a new `[Unreleased]` changelog entry.

### CODE REVIEW

File: frontend/src/lib/types.ts [hunk 1]
```diff
@@ -29,6 +29,8 @@ export interface LiveFrame {
   car_name: string;
   session_best_ms: number;
   prev_best_ms: number; // session best BEFORE the latest lap (-1 if none)
+  delta_ms: number | null; // live gap vs the session-best lap (null = no reference)
+  lap_elapsed_ms: number; // time into the current lap (-1 before the first sample)
```
[the remark this answers is not in the document]
Comment (jbhoorasingh) [reply frontend/src/lib/types.ts:35]: Fixed in b97118d — the comment now reads "null whenever it's unavailable (no reference lap yet, paused/off-track, or past the reference's last sample)".

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: frontend/src/views/LiveView.tsx [hunk 2]
```diff
@@ -135,11 +134,11 @@ function Dashboard({ frame }: { frame: LiveFrame }) {
...
               <Row k="Best" v={formatLapTime(frame.best_lap_ms)} accent />
-              {lastVsBest !== null && (
+              {delta !== null && (
                 <Row
-                  k="Δ best"
-                  v={formatDelta(lastVsBest)}
-                  className={lastVsBest <= 0 ? "text-throttle" : "text-brake"}
+                  k={delta.live ? "Δ best" : "Δ best (last)"}
+                  v={formatDelta(delta.ms)}
+                  className={delta.ms <= 0 ? "text-throttle" : "text-brake"}
```
[the remark this answers is not in the document]
Comment (jbhoorasingh) [reply frontend/src/views/LiveView.tsx:141]: Fixed in b97118d — LiveView now uses the same "Δ best (last lap)" label as the widgets and docs.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
